packages/dialtone-tokens/build-layered.js

```diff
@@ -0,0 +1,255 @@
+ */
+async function main() {
+  console.log('Step 1: Generating full layered tokens for all themes...\n');
+
+  // First, generate all the layered tokens
+  await runLayeredTokens();
+
+  console.log('\nStep 1b: Running postcss to generate composite tokens...\n');
+
+  const outputDir = 'dist/css/layered';
+
+  // Run postcss on files - use original plugin for :root, layered plugin for mode-specific
+  await runPostCss(`${outputDir}/tokens-core.css`, true); // Original plugin for :root
+  await runPostCss(`${outputDir}/tokens-base-colors.css`, false); // Layered plugin for [data-dt-mode]
+  await runPostCss(`${outputDir}/tokens-dp-colors.css`, false); // Layered plugin for [data-dt-mode]
+
+  console.log('Generated composite tokens (typography, shadows)');
+
+  console.log('\nStep 2: Generating optimized override files...\n');
+
+  // Read the base dp theme
+  const dpContent = await fs.readFile(`${outputDir}/tokens-dp-colors.css`, 'utf8');
+  const dpVars = parseCssVariables(dpContent);
+
+  // Get all theme files
+  const files = await fs.readdir(outputDir);
+  const themeFiles = files.filter(f =>
+    f.startsWith('tokens-') &&
+    f.endsWith('-colors.css') &&
+    f !== 'tokens-dp-colors.css' &&
+    f !== 'tokens-base-colors.css',
+  );
+
+  let totalOriginalSize = 0;
+  let totalOptimizedSize = 0;
+
+  for (const themeFile of themeFiles) {
+    const themeName = themeFile.replace('tokens-', '').replace('-colors.css', '');
+
+    // Read theme file
+    const themeContent = await fs.readFile(`${outputDir}/${themeFile}`, 'utf8');
+    const themeVars = parseCssVariables(themeContent);
+
+    // Generate overrides only
+    const overridesCss = generateOverridesCss(dpVars, themeVars, themeName);
+
+    // Write to themes directory
+    const themesDir = `${outputDir}/themes`;
+    await fs.mkdir(themesDir, { recursive: true });
+    const themesFile = `${themesDir}/${themeFile}`;
+    await fs.writeFile(themesFile, overridesCss);
+
+    const originalSize = Buffer.byteLength(themeContent, 'utf8');
+    const optimizedSize = Buffer.byteLength(overridesCss, 'utf8');
+
+    totalOriginalSize += originalSize;
+    totalOptimizedSize += optimizedSize;
+
+    const reduction = ((1 - optimizedSize / originalSize) * 100).toFixed(1);
+    console.log(`${themeName.padEnd(20)} ${(originalSize / 1024).toFixed(2)}KB → ${(optimizedSize / 1024).toFixed(2)}KB (${reduction}% reduction)`);
+  }
+
+  console.log('\n' + '='.repeat(60));
+  console.log(`Total reduction: ${((1 - totalOptimizedSize / totalOriginalSize) * 100).toFixed(1)}%`);
+  console.log(`Original total: ${(totalOriginalSize / 1024).toFixed(2)}KB`);
+  console.log(`Optimized total: ${(totalOptimizedSize / 1024).toFixed(2)}KB`);
+  console.log(`Savings: ${((totalOriginalSize - totalOptimizedSize) / 1024).toFixed(2)}KB`);
+
+  // Step 3: Clean up redundant full theme files (keep only dp, base, core files)
+  console.log('\nStep 3: Cleaning up redundant full theme files...\n');
```

Comment:
sorta confused why we are generating theme files and then deleting them, is this just for testing purposes?